music/src/ddsp/add_reverb.ts

```diff
@@ -0,0 +1,59 @@
+import * as Tone from 'tone';
+import { arrayBufferToAudioBuffer } from './buffer_utils';
+
+export async function addReverb({
+  audioCtx,
+  arrayBuffer,
+  sampleRate,
+}: {
+  audioCtx: AudioContext;
+  arrayBuffer: Float32Array;
+  sampleRate: number;
+}): Promise<Float32Array> {
+  Tone.setContext(audioCtx);
+  let bufferWithReverb;
+  let resampledAudioBuffer = arrayBufferToAudioBuffer(
+    audioCtx,
+    arrayBuffer,
+    sampleRate
+  );
+  let limiter: AudioNode,
+    reverb: AudioNode,
+    filter: AudioNode,
+    player: AudioBufferSourceNode;
+  const renderingPromise = Tone.Offline(() => {
+    // prevent sounds from going too loud
+    limiter = new Tone.Compressor({
+      attack: 0.001,
+      release: 0.001,
+      threshold: -6,
+    }).toDestination();
+    reverb = new Tone.Reverb({
+      wet: 0.3,
+      decay: 3,
+    }).connect(limiter);
+    filter = new Tone.Filter(8000, 'lowpass', -24).connect(reverb);
+    player = new Tone.Player({
+      url: resampledAudioBuffer,
+    }).connect(filter);
+    player.start();
+    // @ts-ignore
+    return reverb.ready;
+  }, arrayBuffer.length / sampleRate);
+  bufferWithReverb = await renderingPromise;
+  const bufferWithReverbData = bufferWithReverb.getChannelData(0);
+  // manual cleanup for safari
+  resampledAudioBuffer = null;
+  // @ts-ignore
```

Comment:
Hmm, why are there so many ignores?

Reply:
@notwaldorf I have a TODO to fix types (which I privately DM'ed Jesse) so apologies for not stating it in this draft PR!  (It's also why this is still a draft PR and not ready for a review)

In this case, I'm actually struggling a little with TFJS and TS where they expect a Tensor in the return of a prediction but our models actually return an array with 2 tensors.